src/crawlee/browsers/playwright_browser_controller.py

```diff
@@ -69,9 +71,21 @@ def is_browser_connected(self) -> bool:
         return self._browser.is_connected()
 
     @override
-    async def new_page(self, page_options: Mapping[str, Any] | None = None) -> Page:
+    async def new_page(
+        self,
+        page_options: MutableMapping[str, Any] | None = None,
+        proxy_info: ProxyInfo | None = None,
+    ) -> Page:
         page_options = page_options or {}
 
+        # If "proxy_info" is provided and no proxy is already set in "page_options", configure the proxy.
+        if proxy_info and 'proxy' not in page_options:
+            page_options['proxy'] = {
+                'server': f'{proxy_info.scheme}://{proxy_info.hostname}:{proxy_info.port}',
```

Comment:
We have multiple libraries that can compose URLs for us, why not use one of them? :slightly_smiling_face:

Reply:
I'm not sure if using HTTPX outside of the `httpx_client` module is a good idea, especially for such trivial operations.

Reply:
Valid point. Maybe there is similar functionality in pydantic, but I'm shooting blind here.

Reply:
this feels like bikeshedding to me, `f'{proxy_info.scheme}://{proxy_info.hostname}:{proxy_info.port}'` looks just fine, lets focus on more important stuff :]